xbmc/addons/addoninfo/AddonInfo.h

```diff
@@ -26,6 +26,18 @@ class CAddonInfo;
 typedef std::shared_ptr<CAddonInfo> AddonInfoPtr;
 typedef std::vector<AddonInfoPtr> AddonInfos;
 
+enum AddonDisabledReason
+{
+  // Special reason for returning all disabled addons.
+  // Only used as an actaul value when an addon is enabled.
+  ADDON_DISABLED_REASON_NONE = 0,
+  ADDON_DISABLED_REASON_UNKNOWN = 1,
+  ADDON_DISABLED_REASON_USER = 2,
+  ADDON_DISABLED_REASON_INCOMPATIBLE = 3,
+  ADDON_DISABLED_REASON_BROKEN = 4,
```

Comment:
Are broken addons automatically disabled, or is the user prompted? I am under the impression that when you mark the addon as broken you are notified and asked if you want to disable the addon. If this is true, is the reason being changed from `ADDON_DISABLED_REASON_BROKEN` to `ADDON_DISABLED_REASON_USER`?